_posts/2019/2019-05-07-react-hooks-what-all-the-fuss-is-about.md

```diff
@@ -0,0 +1,55 @@
+
+And that's where hooks come in. **Hooks allow you to do all the things you previously needed classes for, inside a function-component.** 🎉
+
+## What makes them so great?
+
+Now you might be asking yourself: what's the big deal? You can write your components in a different way. What's so crazy about that?
+
+In some ways you'd be right, there's nothing fundamentally new here. But **React Hooks solve a variety of problems**. I want to hit on a few of them here.
+
+### Improved reusability
+
+Before hooks, React didn't provide a way to "attach" reusable behavior to a component. Instead, patterns emerged that would allow sharing stateful behavior between components. These, however, were burdensome and made the code harder to follow.
+
+With hooks, stateful logic can be extracted into independent pieces of code that are easily shared between components and projects.
+
+### Reduced complexity
+
+When a class-component increases in complexity, it tends to grow into an unmaintainable mess. That's because class-components make your organize your logic in lifecycle methods. As a result, you end up with lifecycle methods that contain a mix of unrelated logic. If you now wanted to take out a specific behavior of your component and move it to a different one, you would have to go look at all the different lifecycle methods and extract only the parts you need and put them somewhere else. This is hard and makes refactoring painful.
+
+The new Hooks API, on the other hand, allows you to separate your component into smaller pieces (hooks), which are responsible for a specific behavior or logic. This not only makes the component easier to understand, but it also lets you easier eject some behavior from a component and put it somewhere else. Whereas before you would have had to look at different lifecycle methods, you can now just take out your hook and put it somewhere else.
+
+### Classes can be confusing
+
+Classes are one of the harder things for React-beginners to understand and master. They tend to be quite verbose and come with some quirks one has to be aware of. And even experienced React developers struggle when to use a class and when to use a function.
+
+With hooks, all these problems fade away. You can do anything you want with a function-component now, and hooks will most likely even encourage you to use better patterns for your components.
+
+**No confusion about how to use classes, and no struggle when to use them.** 🎉
+
+## What does this mean for me?
+
+If you are not working with React, this new API won't affect you much. Except that your developers that do work with React might have a better time coding, and as a result will be more effective at building cool and stable UIs. It probably also means that React is not going away anytime soon, as the React-team keeps innovating, while at the same time provide a great developer experience.
```

Comment:
You rephrased it already yourself while explaining it to me :)

Even if you're not a React developer, and this new React API won't affect you directly, it will affect your colleagues that **do** work with it. And it will affect them positively, as they will be more effective at building cool and stable UIs.
It probably also means that React is not going away anytime soon, as the React team keeps innovating, while at the same time they provide a great developer experience.

Reply:
Thanks!